### ISSUE TITLE
BUG: Log Debug Messages

### ISSUE THREAD

**Issue by SPDurkee** [tnodir/fort#123]
**Log Debug Messages** doesn't seem to do anything. Also there seems to be very little information going to the log. Is there a setting somewhere I am missing or is this a bug?
![image](https://github.com/tnodir/fort/assets/2486134/1560aa95-cd0f-4f16-9a86-cc6d88267e26)

This is all I am getting in the console window
![image](https://github.com/tnodir/fort/assets/2486134/4901a1d9-2403-4aed-8824-334ba7ecb5af)

**Comment by SPDurkee** [comment]
Here is more detail:
I have a portable install located in a folder with valid write permissions.
Very little data is being sent to the log, even with Debug checked.
```
2023-10-08 00:48:18.617 I Fort Firewall v3.9.12
2023-10-08 00:48:18.617 I fortManager: Started as Service Client
2023-10-08 00:52:07.748 I control: Client connected: id: 2 count: 1
2023-10-08 00:52:07.765 I control: Client disconnected: id: 2
2023-10-08 00:52:13.047 I control: Client connected: id: 3 count: 1
2023-10-08 00:52:13.208 I control: Client disconnected: id: 3
2023-10-08 00:52:29.326 I manager.window: Quit due user request
2023-10-08 00:52:30.883 I Fort Firewall v3.9.12
2023-10-08 00:52:30.883 I fortManager: Started as Service Client
2023-10-08 03:36:56.273 I manager.window: Quit due user request
2023-10-08 03:38:21.216 I Fort Firewall v3.9.12
2023-10-08 03:38:21.216 I fortManager: Started as Program
2023-10-08 04:33:24.377 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 05:38:27.513 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 06:43:27.584 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 07:48:27.683 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
```

Please let me know what else I can provide to help troubleshoot this. Thank you!

**Comment by SPDurkee** [comment]
Updated Log Today
```
2023-10-08 03:38:21.216 I Fort Firewall v3.9.12
2023-10-08 03:38:21.216 I fortManager: Started as Program
2023-10-08 04:33:24.377 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 05:38:27.513 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 06:43:27.584 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 07:48:27.683 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 08:53:27.758 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 09:58:27.867 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 11:03:27.955 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
```

**Comment by tnodir** [comment]
Debug messages look fine to me.

OTOH, error message about read-only portable installation should be in the logs/ folder, not console.

**Comment by SPDurkee** [comment]
All the log seems to do is indicate when the program is started and when new versions are checked.

I guess I expected a lot more in the log, such as:
Programs Discovered / Blocked / Allowed
Blocked Connections
Options Changed / Saved (INI File Changes)

For debugging I would think just about every action would be logged so you see what may be causing the problem you are having.

The main reason I would use the log is when an application is being blocked/allowed and I'm trying to track down why it is happening. Finding the rule/cause of the block/allowance can be quite convoluted and time consuming.

**Comment by SPDurkee** [comment]
I assume there is no point in providing any more logs, here is the remaining logs for today.
```
2023-10-08 03:38:21.216 I Fort Firewall v3.9.12
2023-10-08 03:38:21.216 I fortManager: Started as Program
2023-10-08 04:33:24.377 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 05:38:27.513 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 06:43:27.584 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 07:48:27.683 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 08:53:27.758 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 09:58:27.867 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 11:03:27.955 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 12:08:28.013 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 13:13:28.048 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 14:18:28.158 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 15:23:28.225 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 16:28:28.327 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 17:33:28.395 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 18:38:28.466 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 19:43:28.515 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 20:48:28.582 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 21:53:28.656 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
2023-10-08 22:58:28.836 I util.net.netDownloader: NetDownloader: Start: "https://api.github.com/repos/tnodir/fort/releases/latest" ""
```

**Comment by SPDurkee** [comment]
One issue I have noticed is that when **_Show log messages in console_** is checked, timestamps are not included in the console messages.

**Comment by tnodir** [comment]
> For debugging I would think just about every action would be logged so you see what may be causing the problem you are having.

Yes, it'll be helpful to have..

> The main reason I would use the log is when an application is being blocked/allowed and I'm trying to track down why it is happening.

Please use Statistics window: "Blocked Connections" tab. The "Dir." column will show a blocking reason.

**Comment by tnodir** [comment]
> One issue I have noticed is that when Show log messages in console is checked, timestamps are not included in the console messages.

Fixed in b25fd829

**Comment by tnodir** [comment]
> Options Changed / Saved (INI File Changes)

I want to add debug output for these actions, so you can see errors in console.

And then I can close this issue.

**Comment by tnodir** [comment]
Implemented by b529f714.
